src/sentry/search/events/builder/spans_metrics.py

```diff
@@ -22,6 +24,24 @@ def get_field_type(self, field: str) -> Optional[str]:
 
         return None
 
+    def resolve_select(
+        self, selected_columns: Optional[List[str]], equations: Optional[List[str]]
+    ) -> List[SelectType]:
+        if selected_columns and "transaction" in selected_columns:
+            self.has_transaction = True
```

Comment:
We should also add this to `resolve_filter` as well 👀

Reply:
So I'm actually sorta cheating it here cause `resolve_conditions` happens before `resolve_select` it'll call `resolve_metric_index` which is what L38 is doing.

Unfortunately that won't work for the `selected_columns` cause nothing in there depends on order (and i didn't want to make it depend on order) so I have to check the raw column name here instead